File: src/LorAuto/Plugin/Types/PluginBase.cs
using LorAuto.Plugin.Model;

namespace LorAuto.Plugin.Types;

/// <summary>
/// Base class for all plugins
/// </summary>
public abstract class PluginBase : IDisposable
{
    /// <summary>
    /// Gets the information about the plugin.
    /// </summary>
    public abstract PluginInfo PluginInformation { get; }

    /// <summary>
    /// Disposes the resources used by this instance.
    /// </summary>
    public virtual void Dispose()
    {
        GC.SuppressFinalize(this);
    }
}

File: src/LorAuto.SDK.Python/LorAuto/Plugin/PluginBase.py
from abc import ABC, abstractmethod

from LorAuto.Plugin.Model.PluginInfo import PluginInfo


class PluginBase(ABC):
    @property
    @abstractmethod
    def PluginInformation(self) -> PluginInfo:
        """
        Gets the information about the plugin.

        Returns:
            PluginInfo: The information about the plugin.
        """
        pass

    def Dispose(self) -> None:
        """
        Disposes the resources used by this instance.
        """
        pass
